conan/conan.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package conan
+
+import (
+    "github.com/package-url/packageurl-go"
+    "github.com/sirupsen/logrus"
+    "os"
+    "path/filepath"
+    "strings"
+)
+
+var (
+    newline = "\n"
+)
+
+type conanPurlInfo struct {
+    name string
+    version string
+}
+
+type Options struct {
+    Directory  string
+    BinaryName string
+}
+
+type ConanGenerator struct {
+    filepath string
+    logger *logrus.Logger
+}
+
+func New(logger *logrus.Logger, options Options) *ConanGenerator {
+    if options.Directory == "" {
+        options.Directory = "."
```

Comment:
This by intent? Wondering if it will work on Windows (hence the question). Cross platform way to get the current working directory:

`	dir, err := os.Getwd()`

Reply:
oh that's great, i'll change that